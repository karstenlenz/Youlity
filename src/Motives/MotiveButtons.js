import React from 'react'
import styled from 'styled-components'
import Button from '../common/Button'
import InfoOverlay from '../common/InfoOverlay'

export default function MotiveButtons({ handleMotiveClick }) {
  const motiveData = [
    {
      name: 'Anerkennung',
      description: () => {
        return (
          <>
            <p>
              Das Bedürfnis nach Anerkennung bedeutet, Lorem ipsum dolor sit
              amet consectetur, adipisicing elit. Ipsum sit quod accusantium
              voluptatibus tenetur dolor? Necessitatibus, debitis unde
              repellendus non cupiditate doloribus? Accusantium debitis
              temporibus vel aut, consectetur distinctio! Obcaecati eum vel
              natus mollitia voluptatibus ullam quae esse sit tempora.
            </p>
            <ul>
              <li>„Du bist als lorem“</li>
              <li>„Du bist ipsum.“</li>
              <li>„Du siehst dolor aus.“</li>
              <li>„Du bist sit.“</li>
            </ul>
          </>
        )
      },
    },
    {
      name: 'Wichtigkeit',
      description: () => {
        return (
          <>
            <p>
              Das Bedürfnis nach Wichtigkeit bedeutet, Lorem ipsum dolor sit
              amet consectetur, adipisicing elit. Ipsum sit quod accusantium
              voluptatibus tenetur dolor? Necessitatibus, debitis unde
              repellendus non cupiditate doloribus? Accusantium debitis
              temporibus vel aut, consectetur distinctio! Obcaecati eum vel
              natus mollitia voluptatibus ullam quae esse sit tempora.
            </p>
            <ul>
              <li>„Du bist als lorem“</li>
              <li>„Du bist ipsum.“</li>
              <li>„Du siehst dolor aus.“</li>
              <li>„Du bist sit.“</li>
            </ul>
          </>
        )
      },
    },
    {
      name: 'Solidarität',
      description: () => {
        return (
          <>
            <p>
              Das Bedürfnis nach Solidarität bedeutet, Lorem ipsum dolor sit
              amet consectetur, adipisicing elit. Ipsum sit quod accusantium
              voluptatibus tenetur dolor? Necessitatibus, debitis unde
              repellendus non cupiditate doloribus? Accusantium debitis
              temporibus vel aut, consectetur distinctio! Obcaecati eum vel
              natus mollitia voluptatibus ullam quae esse sit tempora.
            </p>
            <ul>
              <li>„Du bist als lorem“</li>
              <li>„Du bist ipsum.“</li>
              <li>„Du siehst dolor aus.“</li>
              <li>„Du bist sit.“</li>
            </ul>
          </>
        )
      },
    },
    {
      name: 'Autonomie',
      description: () => {
        return (
          <>
            <p>
              Das Bedürfnis nach Autonomie bedeutet, Lorem ipsum dolor sit amet
              consectetur, adipisicing elit. Ipsum sit quod accusantium
              voluptatibus tenetur dolor? Necessitatibus, debitis unde
              repellendus non cupiditate doloribus? Accusantium debitis
              temporibus vel aut, consectetur distinctio! Obcaecati eum vel
              natus mollitia voluptatibus ullam quae esse sit tempora.
            </p>
            <ul>
              <li>„Du bist als lorem“</li>
              <li>„Du bist ipsum.“</li>
              <li>„Du siehst dolor aus.“</li>
              <li>„Du bist sit.“</li>
            </ul>
          </>
        )
      },
    },
    {
      name: 'Grenzen',
      description: () => {
        return (
          <>
            <p>
              Das Bedürfnis nach Grenzen bedeutet, Lorem ipsum dolor sit amet
              consectetur, adipisicing elit. Ipsum sit quod accusantium
              voluptatibus tenetur dolor? Necessitatibus, debitis unde
              repellendus non cupiditate doloribus? Accusantium debitis
              temporibus vel aut, consectetur distinctio! Obcaecati eum vel
              natus mollitia voluptatibus ullam quae esse sit tempora.
            </p>
            <ul>
              <li>„Du bist als lorem“</li>
              <li>„Du bist ipsum.“</li>
              <li>„Du siehst dolor aus.“</li>
              <li>„Du bist sit.“</li>
            </ul>
          </>
        )
      },
    },
  ]

  return (
    <ButtonList>
      {motiveData.map((motive) => {
        return (
          <>
            <Button
              key={motive.name}
              onClick={(event) => handleMotiveClick(event, motive.name)}
            >
              {motive.name}
            </Button>
            <InfoOverlay>{motive.description()}</InfoOverlay>
          </>
        )
      })}
    </ButtonList>
  )
}

const ButtonList = styled.section`
  display: flex;
  flex-direction: column;

  button {
    height: 40px;
    margin-bottom: 20px;
  }
`
