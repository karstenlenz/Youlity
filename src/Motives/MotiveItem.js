import React from 'react'
import { Draggable } from 'react-beautiful-dnd'
import styled from 'styled-components/macro'
import InfoOverlay from '../common/InfoOverlay'
import { motiveData } from '../data/motiveData'

export default function ({
  children,
  index,
  isDragDisabled,
  motiveIndex,
  onClick,
  droppableId,
}) {
  return (
    <Draggable
      draggableId={'draggable-' + children}
      index={index}
      isDragDisabled={isDragDisabled}
    >
      {(provided, snapshot) => (
        <MotiveItemStyled
          ref={provided.innerRef}
          {...provided.draggableProps}
          {...provided.dragHandleProps}
          onClick={() => onClick(motiveIndex + 1, droppableId)}
          style={getStyle(provided.draggableProps.style, snapshot)}
          isDragging={snapshot.isDragging}
        >
          {children}
          <PositionedInfoOverlay>
            <h2>{motiveData[motiveIndex].name}</h2>
            <p>{motiveData[motiveIndex].description}</p>
          </PositionedInfoOverlay>
        </MotiveItemStyled>
      )}
    </Draggable>
  )

  function getStyle(style, snapshot) {
    if (!snapshot.isDragging) return {}
    if (!snapshot.isDropAnimating) {
      return style
    }

    return {
      ...style,
      transitionDuration: `0.001s`,
    }
  }
}

const PositionedInfoOverlay = styled(InfoOverlay)`
  position: absolute;
  top: -40px;
  right: -10px;
`

const MotiveItemStyled = styled.div`
  display: inline-block;
  position: relative;
  z-index: var(--front);
  font-family: 'Ubuntu', sans-serif;
  font-size: 1em;
  line-height: 0;
  text-decoration: none;
  text-align: center;
  color: var(--dark-grey);
  background: white;
  box-shadow: var(--primary-shadow);
  width: 100%;
  max-width: 47.5vw;
  height: 58px;
  padding: 29px 0;
  border: ${(props) =>
    props.isDragging ? '1px solid var(--dark-grey)' : 'none'};
  border-radius: 12px;
`
