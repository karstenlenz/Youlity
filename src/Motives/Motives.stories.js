import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import MotivesEntry from './MotivesEntry'

export default {
  title: 'Motive Selection',
  component: MotivesEntry,
}

export const MotivesEntryComponent = () => (
  <Router>
    <MotivesEntry />
  </Router>
)
