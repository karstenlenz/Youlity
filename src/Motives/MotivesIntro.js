import React from 'react'

export default function MotivesIntro() {
  return <></>
}
