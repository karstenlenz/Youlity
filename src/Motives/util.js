import { styleData } from '../data/styleData'

export function evaluateMatchingStyles(userMotives) {
  const userStylesMatch = {}

  styleData.forEach((style) => {
    if (style.motives.includes(userMotives[0])) {
      let matchPoints = 5
      if (userMotives[0] === style.motives[0]) {
        matchPoints += 5
      }
      if (style.motives.includes(userMotives[1])) {
        matchPoints += 2.5
      }
      if (style.motives.includes(userMotives[2])) {
        matchPoints += 2.5
      }
      // compensate for different array lengths / specificity
      matchPoints = matchPoints * 5 + 30 / style.motives.length
      userStylesMatch[style.id] = matchPoints
    } else {
      userStylesMatch[style.id] = 0
    }
  })
  const sortedUserStyles = Object.keys(userStylesMatch).sort(
    (a, b) => userStylesMatch[b] - userStylesMatch[a]
  )
  const sortedUserStylesAsInts = sortedUserStyles.map((id) => parseInt(id))
  return sortedUserStylesAsInts
}
