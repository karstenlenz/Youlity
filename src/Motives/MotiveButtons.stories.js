import React from 'react'
import MotiveButtons from './MotiveButtons'

export default {
  title: 'Motive Buttons',
  component: MotiveButtons,
}

export const MotiveButtonsComponent = () => (
  <MotiveButtons handleMotiveClick={() => ''} />
)
