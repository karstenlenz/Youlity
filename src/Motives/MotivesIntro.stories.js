import React from 'react'
import MotivesIntro from './MotivesIntro'
import { BrowserRouter as Router } from 'react-router-dom'

export default {
  title: 'Motives/Motives Intro',
  component: MotivesIntro,
}

export const MotivesIntroComponent = () => (
  <Router>
    <MotivesIntro />
  </Router>
)
