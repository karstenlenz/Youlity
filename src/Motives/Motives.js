import React, { useEffect, useState } from 'react'
import { useHistory } from 'react-router-dom'
import MotiveButtons from './MotiveButtons'
import { evaluateMatchingStyles } from './util'
import HeadlineUnderline from '../common/HeadlineUnderline'
import styled from 'styled-components'

export default function Motives() {
  const [userMotives, setUserMotives] = useState([])
  const history = useHistory()

  useEffect(() => {
    if (userMotives.length === 3) {
      const userPersonalityStyles = evaluateMatchingStyles(userMotives)
      const questionnaireUrl =
        '/questionnaire/' + userPersonalityStyles[0] + userPersonalityStyles[1]
      setUserMotives([])
      history.push(questionnaireUrl)
    }
  }, [userMotives, history])

  return (
    <>
      <img alt="" src="/img/motives-intro.svg" />
      <HeadlineUnderline>
        <h1>Schritt 1: Bedürfnisse</h1>
      </HeadlineUnderline>
      <SmallH2>
        Wählen Sie bitte intuitiv die drei Bedürfnisse aus, die für Sie am
        Wichtigsten sind. Entscheiden Sie sich für die Begriffe, die Sie zuerst
        ansprechen.
      </SmallH2>
      <ol>
        <li> {userMotives[0]} </li>
        <li> {userMotives[1]} </li>
        <li> {userMotives[2]} </li>
      </ol>
      <MotiveButtons handleMotiveClick={handleMotiveClick} />
    </>
  )

  function handleMotiveClick(event, motive) {
    const button = event.target
    button.disabled = true
    setUserMotives([...userMotives, motive])
  }
}

const SmallH2 = styled.h4``
