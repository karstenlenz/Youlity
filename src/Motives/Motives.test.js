import { render } from '@testing-library/react'
import React from 'react'
import Motives from './Motives'
import { evaluateMatchingStyles } from './util'

describe('The Motives component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Motives userMotives={[]} setUserMotives={() => true} />
    )
    expect(container).toMatchSnapshot()
  })
})

describe('evaluateStyles', () => {
  it('matches the narcissistic style with correct match', () => {
    const testMotives = [
      'Anerkennung',
      'Autonomie',
      'Solidarit채t',
      'Wichtigkeit',
    ]
    const result = evaluateMatchingStyles(testMotives)
    expect(result[0]).toEqual(1)
  })

  it('also matches the narcissistic style with mixed-up order', () => {
    const testMotives = [
      'Autonomie',
      'Anerkennung',
      'Wichtigkeit',
      'Solidarit채t',
    ]
    const result = evaluateMatchingStyles(testMotives)
    expect(result[0]).toEqual(1)
  })

  it('also matches the histrionic style with correct order', () => {
    const testMotives = ['Wichtigkeit', 'Solidarit채t', 'Verl채sslichkeit']
    const result = evaluateMatchingStyles(testMotives)
    expect(result).toEqual([2, 6, 9, 1, 7, 4, 3, 5, 8])
  })
})
