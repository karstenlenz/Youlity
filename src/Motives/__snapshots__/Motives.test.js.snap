// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Motives component renders correctly 1`] = `
<div>
  <div
    id="root"
  >
    <section
      class="HeadlineUnderline__HeadlineContainer-sc-1reeb6v-1 tAJUt"
    >
      <div
        class="HeadlineUnderline__ChildContainer-sc-1reeb6v-0 cbzWsz"
      >
        <h1>
          Schritt 1: Bedürfnisse
        </h1>
      </div>
      <svg
        class="HeadlineUnderline__UnderlineStyled-sc-1reeb6v-2 dVDSEH"
      >
        underline.svg
      </svg>
    </section>
    <h2
      class="MotivesEntry__SmallH2-zdirr-0 iUHsxQ"
    >
      Wählen Sie bitte intuitiv die drei Bedürfnisse aus, die für Sie am Wichtigsten sind. Sie können die Bedürfnisse hin- und herziehen oder anklicken.
    </h2>
    <section
      class="MotivesEntry__MotiveBG-zdirr-1 hmGxxG"
    >
      <div
        class="UserMotiveDropzone__DropzoneBG-jv6lv-0 cFWTRG"
      >
        <div
          class="HideElement-m9nwk9-0 hYebHX"
        />
      </div>
      <div
        class="UserMotiveDropzone__DropzoneBG-jv6lv-0 cFWTRG"
      >
        <div
          class="HideElement-m9nwk9-0 hYebHX"
        />
      </div>
      <div
        class="UserMotiveDropzone__DropzoneBG-jv6lv-0 cFWTRG"
      >
        <div
          class="HideElement-m9nwk9-0 hYebHX"
        />
      </div>
    </section>
    <div
      class="MotivesList__MotiveContainer-sc-935jjf-0 jbKbTJ"
    >
      <div
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        class="MotiveItem__MotiveItemStyled-sc-17qm44j-1 lnZpnZ"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="draggable-Anerkennung"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="draggable-Anerkennung"
        draggable="false"
        role="button"
        tabindex="0"
      >
        Anerkennung
        <button
          class="Button__BtnStyled-sc-129yfz5-0 ckSRcb InfoOverlay__InfoButton-sc-17974c-0 eQphtg MotiveItem__PositionedInfoOverlay-sc-17qm44j-0 cxSIYu"
          width="100"
        >
          i
        </button>
      </div>
      <div
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        class="MotiveItem__MotiveItemStyled-sc-17qm44j-1 lnZpnZ"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="draggable-Wichtigkeit"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="draggable-Wichtigkeit"
        draggable="false"
        role="button"
        tabindex="0"
      >
        Wichtigkeit
        <button
          class="Button__BtnStyled-sc-129yfz5-0 ckSRcb InfoOverlay__InfoButton-sc-17974c-0 eQphtg MotiveItem__PositionedInfoOverlay-sc-17qm44j-0 cxSIYu"
          width="100"
        >
          i
        </button>
      </div>
      <div
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        class="MotiveItem__MotiveItemStyled-sc-17qm44j-1 lnZpnZ"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="draggable-Solidarität"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="draggable-Solidarität"
        draggable="false"
        role="button"
        tabindex="0"
      >
        Solidarität
        <button
          class="Button__BtnStyled-sc-129yfz5-0 ckSRcb InfoOverlay__InfoButton-sc-17974c-0 eQphtg MotiveItem__PositionedInfoOverlay-sc-17qm44j-0 cxSIYu"
          width="100"
        >
          i
        </button>
      </div>
      <div
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        class="MotiveItem__MotiveItemStyled-sc-17qm44j-1 lnZpnZ"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="draggable-Autonomie"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="draggable-Autonomie"
        draggable="false"
        role="button"
        tabindex="0"
      >
        Autonomie
        <button
          class="Button__BtnStyled-sc-129yfz5-0 ckSRcb InfoOverlay__InfoButton-sc-17974c-0 eQphtg MotiveItem__PositionedInfoOverlay-sc-17qm44j-0 cxSIYu"
          width="100"
        >
          i
        </button>
      </div>
      <div
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        class="MotiveItem__MotiveItemStyled-sc-17qm44j-1 lnZpnZ"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="draggable-Grenzen"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="draggable-Grenzen"
        draggable="false"
        role="button"
        tabindex="0"
      >
        Grenzen
        <button
          class="Button__BtnStyled-sc-129yfz5-0 ckSRcb InfoOverlay__InfoButton-sc-17974c-0 eQphtg MotiveItem__PositionedInfoOverlay-sc-17qm44j-0 cxSIYu"
          width="100"
        >
          i
        </button>
      </div>
      <div
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        class="MotiveItem__MotiveItemStyled-sc-17qm44j-1 lnZpnZ"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="draggable-Verlässlichkeit"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="draggable-Verlässlichkeit"
        draggable="false"
        role="button"
        tabindex="0"
      >
        Verlässlichkeit
        <button
          class="Button__BtnStyled-sc-129yfz5-0 ckSRcb InfoOverlay__InfoButton-sc-17974c-0 eQphtg MotiveItem__PositionedInfoOverlay-sc-17qm44j-0 cxSIYu"
          width="100"
        >
          i
        </button>
      </div>
      <div
        class="HideElement-m9nwk9-0 hYebHX"
      />
    </div>
    <a
      class="FloatingButtonContainer__ButtonContainer-slkv4p-0 hTXSTP"
      href="/"
    >
      <button
        class="Button__BtnStyled-sc-129yfz5-0 jyIaa-D"
        disabled=""
        width="100"
      >
        Fragebogen starten
      </button>
    </a>
    <div
      class="FloatingButtonContainer__BottomSpacer-slkv4p-1 hgDFeh"
    />
  </div>
</div>
`;
